src/sentry/api/authentication.py

```diff
@@ -292,37 +341,59 @@ def accepts_auth(self, auth: list[bytes]) -> bool:
         return not token_str.startswith(SENTRY_ORG_AUTH_TOKEN_PREFIX)
 
     def authenticate_token(self, request: Request, token_str: str) -> tuple[Any, Any]:
-        token = SystemToken.from_request(request, token_str)
-        try:
-            token = (
-                token
-                or ApiToken.objects.filter(token=token_str)
-                .select_related("user", "application")
-                .get()
-            )
-        except ApiToken.DoesNotExist:
+        user: AnonymousUser | RpcUser | None = AnonymousUser()
+
+        token: SystemToken | ApiTokenReplica | ApiToken | None = SystemToken.from_request(
+            request, token_str
+        )
+        application_is_active = True
+
+        if not token:
+            if SiloMode.get_current_mode() == SiloMode.REGION:
+                atr = token = ApiTokenReplica.objects.filter(token=token_str).last()
+                if not atr:
+                    raise AuthenticationFailed("Invalid token")
+                user = user_service.get_user(user_id=atr.user_id)
+                application_is_active = atr.application_is_active
+            else:
+                try:
+                    at = token = (
+                        ApiToken.objects.filter(token=token_str)
+                        .select_related("user", "application")
+                        .get()
+                    )
+                except ApiToken.DoesNotExist:
+                    raise AuthenticationFailed("Invalid token")
+                user = at.user
+                application_is_active = at.application is None or at.application.is_active
+        elif isinstance(token, SystemToken):
+            user = token.user
+
+        if not token:
             raise AuthenticationFailed("Invalid token")
 
         if token.is_expired():
             raise AuthenticationFailed("Token expired")
 
-        if not token.user.is_active:
+        if user and hasattr(user, "is_active") and not user.is_active:
             raise AuthenticationFailed("User inactive or deleted")
 
-        if token.application and not token.application.is_active:
+        if not application_is_active:
```

Comment:
Perhaps we could extract `token.application` into a local var, and then be able to do `if application and not application.is_active` to avoid the incorrect local var.

Reply:
probably yes, will refactor.

Reply:
took another pass on this -- I inverted the condition to make it easier.  The reason why extracting application and using is_active doesn't work great is that ApiTokenReplica has the is_application_active attribute and not a reference to the application directly.  This is because in region mode an application reference is only obtainable either via RPC or replicating that entire object (which seemed overkill).